packages/contracts-bedrock/scripts/upgrades/v1.8.0-permissioned/DeployUpgrade.s.sol

```diff
@@ -0,0 +1,404 @@
+        console.log("Transferring ownership of permissioned DelayedWETH proxy");
+        Proxy prox = Proxy(payable(address(weth)));
+        prox.changeAdmin(proxyAdmin);
+    }
+
+    /// @notice Transfers ownership of the DisputeGameFactory proxy to the ProxyAdmin and transfers
+    ///         ownership of the underlying DisputeGameFactory contract to the SystemOwnerSafe.
+    function transferDGFOwnershipFinal() internal broadcast {
+        // Transfer ownership of DisputeGameFactory to SystemOwnerSafe.
+        console.log("Transferring ownership of underlying DisputeGameFactory");
+        IDisputeGameFactory dgf = IDisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
+        dgf.transferOwnership(systemOwnerSafe);
+
+        // Transfer ownership of DisputeGameFactory proxy to ProxyAdmin.
+        console.log("Transferring ownership of DisputeGameFactory proxy");
+        Proxy prox = Proxy(payable(address(dgf)));
+        prox.changeAdmin(proxyAdmin);
+    }
+
+    /// @notice Transfers ownership of the AnchorStateRegistry proxy to the ProxyAdmin.
+    function transferAnchorStateOwnershipFinal() internal broadcast {
+        // Transfer ownership of AnchorStateRegistry proxy to ProxyAdmin.
+        console.log("Transferring ownership of AnchorStateRegistry proxy");
+        IAnchorStateRegistry asr = IAnchorStateRegistry(mustGetAddress("AnchorStateRegistryProxy"));
+        Proxy prox = Proxy(payable(address(asr)));
+        prox.changeAdmin(proxyAdmin);
+    }
+
+    /// @notice Checks that the deployed system is configured correctly.
+    function postDeployAssertions() internal view {
+        Types.ContractSet memory contracts = _proxies();
+        contracts.OptimismPortal2 = mustGetAddress("OptimismPortal2");
+
+        // Ensure that `useFaultProofs` is set to `true`.
+        assertTrue(cfg.useFaultProofs(), "DeployUpgrade: useFaultProofs is not set to true");
+
+        // Verify that the DGF is owned by the ProxyAdmin.
+        address dgfProxyAddr = mustGetAddress("DisputeGameFactoryProxy");
+        assertEq(
+            address(uint160(uint256(vm.load(dgfProxyAddr, Constants.PROXY_OWNER_ADDRESS)))),
+            proxyAdmin,
+            "DeployUpgrade: DGF is not owned by ProxyAdmin"
+        );
+
+        // Verify that permissioned DelayedWETH is owned by the ProxyAdmin.
+        address soyWethProxyAddr = mustGetAddress("PermissionedDelayedWETHProxy");
+        assertEq(
+            address(uint160(uint256(vm.load(soyWethProxyAddr, Constants.PROXY_OWNER_ADDRESS)))),
+            proxyAdmin,
+            "DeployUpgrade: Permissioned DelayedWETH is not owned by ProxyAdmin"
+        );
+
+        // Run standard assertions.
+        ChainAssertions.checkDisputeGameFactory(contracts, systemOwnerSafe, true);
+        ChainAssertions.checkPermissionedDelayedWETH(contracts, cfg, true, systemOwnerSafe);
+        ChainAssertions.checkOptimismPortal2(contracts, cfg, false);
```

Comment:
Worth also checking the expected game type for `CANNON_PERMISSIONED`. `ChainAssertions` doesn't do this.

Reply:
Good call, added.